### ISSUE TITLE
Further background research: Secure storage

### ISSUE THREAD

**Issue by zven** [sitcomlab/simport-learning-app#28]

**Comment by zven** [comment]
@felixerdy Norwin's background research:

- [spatialite](https://open.spotify.com/track/61Ds6lcmVUFa01J4DyH85N) (an postgis-like sqlite extension) is a good fit.
  - there are patches that make it available on mobile platforms: [android](https://github.com/geopaparazzi/libjsqlite-spatialite-android/wiki/Android_5.0.0.mk) (well maintained), [iOS](https://github.com/smellman/libspatialite-ios) (not so well maintained)
- for integration of sqlite into ionic/capacitor, there is a ready-made [plugin](https://github.com/capacitor-community/sqlite)
  - it is based on [sqlcipher,](https://github.com/sqlcipher/android-database-sqlcipher) a sqlite extension supporting encryption
  - to make it work with spatialite, [some work from us would be required](https://github.com/capacitor-community/sqlite/issues/45).
  - we need to check the requirements of inference strategies to see if this fast-to-get-started-lib (without spatial) can work for our prototype
- h2 should be [possible as well](http://h2database.com/html/tutorial.html#android), though there is almost no docs on integrating that into mobile apps.
- if we loosen the requirement of data-remains-with-users device (i don't see that happening..?) plenty other options become available. this is feasible with proper data encryption only, and might be a good test-case for our LP-preserving UI components later on

**Comment by felixerdy** [comment]
In the end I decided to use the [Capacitor sqlite plugin](https://github.com/capacitor-community/sqlite) v2.9.x which seems like the best documented and up to date implementation. When / if we really rely on spatial queries we might fork the plugin and [swap sqlite with spatiallite](https://github.com/capacitor-community/sqlite/issues/45). 

I also came across [Cordova-spatialite-storage](https://github.com/DisyInformationssysteme/Cordova-spatialite-storage) which however isn't well documented and I haven't found any encryption methods.
